componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionActivity.java

```diff
@@ -357,7 +351,12 @@ public void onProceedToAnalysisScreen(@NonNull Document document) {
         // analysis didn't complete or
         // the user rotated the image
         getSingleDocumentAnalyzer().removeListener();
-        showFragment(getAnalysisFragment(document));
+        pushFragment(getAnalysisFragment(document));
+    }
+
+    private void showNoResultsScreen(final @NonNull Document document) {
+        getFragmentManager().popBackStack(getCameraFragment().getClass().getSimpleName(), 0);
```

Comment:
Just a question here. I do not remember well about the popBackStack method but this is done asynchronously, right? So, will be there any issue if next fragment is pushed just after this line?
I see that after that inside `pushFragment`, the new `fragment`is added, so in case that there would be any issue concat both or use `popBackStackImmediate()` instead.

Reply:
The FragmentManager works with transactions so popping and showing the new fragment will be scheduled and executed in order.